subgraphs/uniswap-forks/protocols/ubeswap/src/common/handlers/handleReward.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  BigInt,
+  ethereum,
+  Address,
+} from "@graphprotocol/graph-ts";
+import { LiquidityPool, Token, _HelperStore } from "../../../../../generated/schema";
+import { UniswapV2Pair } from "../../../../../generated/templates/StakingRewards/UniswapV2Pair";
+import {
+  INT_ZERO,
+  DEFAULT_DECIMALS,
+  BIGINT_ZERO,
+  BIGDECIMAL_ZERO
+} from "../../../../../src/common/constants";
+import { getOrCreateToken } from "../../../../../src/common/getters";
+import {
+  findNativeTokenPerToken,
+  updateNativeTokenPriceInUSD,
+} from "../../../../../src/price/price";
+import { convertTokenToDecimal } from "../../../../../src/common/utils/utils";
+
+export function handleStakedImpl(
+  event: ethereum.Event,
+  amount: BigInt
+): void {
+  // Return if pool does not exist
+  let pool = LiquidityPool.load(event.address.toHexString());
+  if (!pool) {
+    return;
+  }
+  let stakeToken = getOrCreatePair(pool.inputTokens[0]);
+
+  pool.stakedOutputTokenAmount = pool.stakedOutputTokenAmount!.plus(amount);
+  pool.inputTokenBalances[0] = pool.stakedOutputTokenAmount!;
```

Comment:
Why are you storing the staked token balance in the input tokens as well? To me, it make more sense to just leave the storage of the token pairs in the inputTokenBalances and then store the staked amount in the stakedOutputTokenAmount.